### PR TITLE
Batching improvements

### PR DESCRIPTION
**Pull request by kgodlewski**
Key points:
 * Have `AggregatingQueue` wait a predefined time for incoming operations, to maximize batch size
 * `InternalQueueFeederThread` blocks on the queue with a timeout, reducing latency

### CODE REVIEW

File: src/neptune_scale/core/components/queue_element.py [hunk 1]
```diff
@@ -22,8 +22,8 @@ class SingleOperation(NamedTuple):
     timestamp: float
     # Protobuf serialized (RunOperation)
     operation: bytes
-    # Whether the operation is metadata update or not (run creation)
-    is_metadata_update: bool
+    # Whether the operation is batchable. Eg. metadata updates are, while run creations are not.
+    is_batchable: bool
```
Comment (Raalsky) [inline src/neptune_scale/core/components/queue_element.py:26]: Good one 👍

File: src/neptune_scale/core/components/sync_process.py [hunk 2]
```diff
@@ -367,7 +367,11 @@ def commit(self) -> None:
 
     def work(self) -> None:
         try:
-            while (operation := self.get_next()) is not None:
+            while not self._is_interrupted():
```
Comment (Raalsky) [inline src/neptune_scale/core/components/sync_process.py:370]: With Python 3.8 assumption, I think it's possible to to use `while not self._is_interrupted() and (operation := self.get_next()):` instead

File: src/neptune_scale/core/components/aggregating_queue.py [hunk 3]
```diff
@@ -47,37 +50,46 @@ def put_nowait(self, element: SingleOperation) -> None:
...
+        wait_remaining = self._wait_time
 
-        while (element := self._get_next()) is not None:
-            elements_in_batch += 1
+        # Pull operations off the queue until we either reach the maximum size, or
+        # the specified wait time has passed. This way we maximize the potential batch size.
+        while wait_remaining > 0:
+            t0 = time.monotonic()
+
+            if elements_in_batch + 1 > self._max_elements_in_batch:
```
Comment (Raalsky) [inline src/neptune_scale/core/components/aggregating_queue.py:84]:
```suggestion
            if elements_in_batch >= self._max_elements_in_batch:
```

File: src/neptune_scale/core/components/aggregating_queue.py [hunk 4]
```diff
@@ -118,10 +130,12 @@ def get_nowait(self) -> BatchedOperations:
                 batch_timestamp = element.timestamp
 
                 self.commit()
-        else:
-            logger.debug("Batch closed due to queue being empty")
+
+            t1 = time.monotonic()
```
Comment (Raalsky) [inline src/neptune_scale/core/components/aggregating_queue.py:134]: You can just store `t0` and check the remaining time in the condition of `while` directly

File: src/neptune_scale/core/components/sync_process.py (hunk 2 above)
Comment (kgodlewski) [reply src/neptune_scale/core/components/sync_process.py:370]: I don't want to break the loop though, just keep blocking on the queue without any sleeps between calls to `work()`.

**Pull request merged**